### PR TITLE
feat: Improved AWS backup notification readability

### PR DESCRIPTION
**Pull request by falense**
## Description

When using this module with AWS backup events the resulting message is poorly formatted due to the unstructured nature of the message sent by AWS backup. This PR does a best effort parsing of the message in order to improve readability for the user.

### Before
Screenshot from 2024-04-03 08-08-24

### After
Screenshot from 2024-04-03 08-07-47

## Breaking Changes

This should not cause any breaking changes and only affects AWS backup messages

## How Has This Been Tested?

We have been running on our test environment for a couple of weeks now.

### CODE REVIEW

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Looks good. Could you please update tests like described [here](https://github.com/terraform-aws-modules/terraform-aws-notify-slack/tree/master/functions#supporting-additional-events) also?

**Comment by falense** [comment]
I have added a test according to the examples. Did I do it right? I had some troubles with the somewhat older Python3.8 version so I bumped it to 3.10 to be able to generate the snapshot.

**Comment by falense** [comment]
I saw that there were more tests. I have corrected the style errors listed below. Is there anything else I can do to get this merged? 

> ./notify_slack.py:290:35: E711 comparison to None should be 'if cond is None:'
./notify_slack.py:298:38: E711 comparison to None should be 'if cond is None:'
./notify_slack.py:307:52: E203 whitespace before ':'

**Comment by bryantbiggs** [comment]
you'll need to get the CI checks to pass - unfortunately, you have disabled our ability to update your branch so you will have to do these updates on your end

**Comment by falense** [comment]
I have added you both as maintainers on the fork. Maybe that will do the trick?

**Comment by antonbabenko** [comment]
Maybe, I will take a look in a few hours today.

**Comment by falense** [comment]
Brainstormed this with some colleagues. Black suggests adding ignoring  E203 and E701 as a minimal config. See https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#minimal-configuration. I have added the change to the PR.

File: functions/notify_slack.py [hunk 1]
```diff
@@ -266,6 +266,86 @@ def format_aws_health(message: Dict[str, Any], region: str) -> Dict[str, Any]:
...
+
+def format_aws_backup(message: str) -> Dict[str, Any]:
+    """
+    Format AWS Backup event into Slack message format
+
+    :params message: SNS message body containing AWS Backup event
+    :returns: formatted Slack message payload
+    """
+
+    fields:list[Dict[str, Any]] = []
```
Comment (antonbabenko) [inline functions/notify_slack.py:325]:
```suggestion
    fields: list[Dict[str, Any]] = []
```

**Comment by antonbabenko** [comment]
I looked at it, but you still need to fix the failing unit tests and formatting to make GitHub Actions green.

**Review by bryantbiggs — APPROVED** [review]
Added in all of the notification examples from here https://docs.aws.amazon.com/aws-backup/latest/devguide/backup-notifications.html to capture in the snapshots. Not sure why the snapshottest library decided to reformat everything - thats rather unfortunate and noisy but all looks good!

cc @antonbabenko

**Review by antonbabenko — APPROVED** [review]

**Pull request merged**